plugins/plugin-chart-table/src/buildQuery.ts

```diff
@@ -43,7 +45,13 @@ export function getQueryMode(formData: TableChartFormData) {
   return hasRawColumns ? QueryMode.raw : QueryMode.aggregate;
 }
 
-export default function buildQuery(formData: TableChartFormData) {
+type Hooks = {
+  setDataMask?: SetDataMaskHook;
+  cachedChanges: any;
+  setCachedChanges: (newChanges: any) => void;
+};
```

Comment:
Should this typing live in `@superset-ui/core`?

Since you already have the`BuildQueryFunction` generic type, you can just do:

```ts
export cons buildQuery: BuildQuery<TableChartFormData>: (formData, { hooks }) => {
  // ...  hooks would be already typed
};
```

Reply:
I updated now this type to `BuildQuery` type